SA3/qa_web/views.py

```diff
@@ -79,12 +79,25 @@ def questions(request):
 
 def answers(request, id_):
     q = get_object_or_404(Questions, pk=id_)
-    if request.method == 'POST':
+    if request.method == 'POST' and 'answer_form' in request.POST: #Update's database when somebody answers a question
         form = AnswersForm(request.POST)
-        if form.is_valid():
+        if form.is_valid(): 
             Answers.objects.create(content=request.POST['content'], owner=request.user, question=q)
-    q_answers = Answers.objects.filter(question=q)
-    return render(request, 'qa_web/answerspage.html', {'currentQuestion': q, 'answers': q_answers})
+    elif request.method == 'POST' and 'deselect' in request.POST:  #Update's database when somebody deselects best answer.
+        updateAnswer = Answers.objects.get(correct_answer=True)
```

Comment:
Here you needed to specify the corresponding question by doing `.objects.filter(question=q, correct_answer=True)`. But it's probably just something you forgot because you do it in line `91` already. I will include this fix in my commit so it's all good :)